collectors/python.d.plugin/logind/logind.chart.py

```diff
@@ -3,83 +3,165 @@
-    'seats',
 ]
 
 CHARTS = {
-    'sessions': {
-        'options': [None, 'Logind Sessions', 'sessions', 'sessions', 'logind.sessions', 'stacked'],
+    'session_types': {
+        'options': [None, 'Logind Session By Type', 'session_types', 'session_types', 'logind.session_types', 'stacked'],
         'lines': [
             ['sessions_graphical', 'Graphical', 'absolute', 1, 1],
             ['sessions_console', 'Console', 'absolute', 1, 1],
-            ['sessions_remote', 'Remote', 'absolute', 1, 1]
+            ['sessions_remote', 'Remote', 'absolute', 1, 1],
+            ['sessions_other', 'Other', 'absolute', 1, 1]
         ]
     },
-    'users': {
-        'options': [None, 'Logind Users', 'users', 'users', 'logind.users', 'stacked'],
+    'session_states': {
+        'options': [None, 'Logind Sessions By State', 'session_states', 'session_states', 'logind.session_states', 'stacked'],
         'lines': [
-            ['users_graphical', 'Graphical', 'absolute', 1, 1],
-            ['users_console', 'Console', 'absolute', 1, 1],
-            ['users_remote', 'Remote', 'absolute', 1, 1]
+            ['sessions_online', 'Online', 'absolute', 1, 1],
+            ['sessions_active', 'Active', 'absolute', 1, 1],
+            ['sessions_closing', 'Closing', 'absolute', 1, 1]
         ]
     },
-    'seats': {
-        'options': [None, 'Logind Seats', 'seats', 'seats', 'logind.seats', 'line'],
+    'users': {
+        'options': [None, 'Logind Users', 'users', 'users', 'logind.users', 'stacked'],
         'lines': [
-            ['seats', 'Active Seats', 'absolute', 1, 1]
+            ['users_offline', 'Offline', 'absolute', 1, 1],
+            ['users_lingering', 'Lingering', 'absolute', 1, 1],
+            ['users_online', 'Online', 'absolute', 1, 1],
+            ['users_active', 'Active', 'absolute', 1, 1],
+            ['users_closing', 'Closing', 'absolute', 1, 1]
         ]
-    }
+    },
 }
 
 
-class Service(ExecutableService):
+class Service(SimpleService):
     def __init__(self, configuration=None, name=None):
-        ExecutableService.__init__(self, configuration=configuration, name=name)
+        SimpleService.__init__(self, configuration=configuration, name=name)
         self.order = ORDER
         self.definitions = CHARTS
-        self.command = LOGINCTL_COMMAND
+        self.bus_name = configuration.get('bus_name', BUS_NAME)
+        self.obj_path = configuration.get('object_path', OBJ_PATH)
+        self.bus = dbus.SystemBus()
+        self.mgr = None
+        self.iface = None
+
+        try:
+            self.mgr = self.bus.get_object(self.bus_name, self.obj_path)
+            self.iface = dbus.Interface(self.mgr, MGR_IFACE)
+        except dbus.DBusException:
+            pass
+
+    def check(self):
+        if not HAVE_DBUS:
+            self.error('dbus-python is not available')
+            return False
+
+        if self.mgr is None:
+            self.error('Unable to connect to ' + self.bus_name + ' service')
```

Comment:
```suggestion
            self.error('Unable to connect to {} service'.format(self.bus_name))
```

I think format is better than string concat

Reply:
Quite so.

It’s been far too long since I did anything in Python, and forgot about formatting other than f-strings (which I figured we need to not use here, because they’re foo new).